clients/tests/test_gebsr2csr.cpp

```diff
@@ -0,0 +1,215 @@
+/* ************************************************************************
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * ************************************************************************ */
+
+#include "testing_gebsr2csr.hpp"
+#include "utility.hpp"
+
+#include <gtest/gtest.h>
+#include <hipsparse.h>
+#include <string>
+#include <vector>
+
+typedef std::
+    tuple<int, int, int, int, hipsparseIndexBase_t, hipsparseIndexBase_t, hipsparseDirection_t>
+        gebsr2csr_tuple;
+typedef std::
+    tuple<int, int, hipsparseIndexBase_t, hipsparseIndexBase_t, hipsparseDirection_t, std::string>
+        gebsr2csr_bin_tuple;
+
+// Random matrices
+int gebsr2csr_M_range[]             = {-1, 0, 872, 21453};
+int gebsr2csr_N_range[]             = {-3, 0, 623, 29285};
+int gebsr2csr_row_block_dim_range[] = {-1, 0, 1, 2, 4, 8};
+int gebsr2csr_col_block_dim_range[] = {-1, 0, 1, 2, 4, 8};
+
+hipsparseIndexBase_t gebsr2csr_csr_base_range[] = {HIPSPARSE_INDEX_BASE_ZERO};
+
+hipsparseIndexBase_t gebsr2csr_bsr_base_range[] = {HIPSPARSE_INDEX_BASE_ONE};
+
+hipsparseDirection_t gebsr2csr_dir_range[] = {HIPSPARSE_DIRECTION_ROW, HIPSPARSE_DIRECTION_COLUMN};
+
+// Matrices from files (float and double)
+int gebsr2csr_row_block_dim_range_bin[] = {5};
```

Comment:
Is it possible to add more tests with different row and col dimensions?

Reply:
I would at least make them different if it takes too long to run with more tests

Reply:
it is testing with different row and col dimensions, and I also want equal dimensions. That's why I took the same range but reduced.

Reply:
good catch

Reply:
fixed.